src/Game.java

```diff
@@ -1,45 +1,119 @@
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 import java.util.Scanner;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 
 public class Game {
-    public static void main(String[] args) {
-        System.out.println("Welcome to PPTHangman!");
-        System.out.println("Please enter the number of players:");
 
-        Scanner scanner = new Scanner(System.in);
+  public static Map<String, Integer> leaderboardToSet(List<Player> allPlayers) {
+    return allPlayers.stream()
+        .collect(Collectors.toMap(Player::toString, Player::getScore));
+  }
 
-        int numPlayers = Integer.parseInt(scanner.nextLine());
-        List<Player> allPlayers = new ArrayList<>();
+  public static void main(String[] args) {
+    System.out.println("Welcome to PPTHangman!");
+    System.out.println("Please enter the number of players:");
 
-        for (int i = 0; i < numPlayers; i++) {
-            System.out.println("Please enter the name of the " + (i + 1) + "th player");
-            allPlayers.add(new Player(scanner.nextLine()));
+    Scanner scanner;
+    int numPlayers;
+    String numPlayersString;
+
+    while (true) {
+      scanner = new Scanner(System.in);
+      numPlayersString = scanner.nextLine();
+
+      if (numPlayersString.length() == 0){
+        System.out.println("Invalid input, please insert a number:");
+      } else {
+        numPlayers = Integer.parseInt(numPlayersString);
+        if (numPlayers > 1) {
+          break;
+        } else if (numPlayers == 1) {
+          System.out.println(
+              "You cannot play alone! Please enter a number greater than 1:");
+        } else {
+          System.out.println("Please enter a number greater than 1:");
+        }
+      }
+    }
+
+    List<Player> allPlayers = new ArrayList<>();
+
+    for (int i = 0; i < numPlayers; i++) {
+      System.out.println("Please enter the name of player " + (i + 1) + ":");
+      String playerName;
+      while (true) {
```

Comment:
Hmm, this is already largely similar functionality to when you asked the user for the number of players. Now imagine if were taking even more parameters than just player numbers and names, say we had 30 different settings to ask before every game. Would you want to have 30 different while loops within a massive main function?

Try to think of a neat solution to this problem. Maybe you could define a function with a signature like `String askUntilCondition(String message, Function<String, Boolean> condition)`which uses a do-while loop, to keep asking the user for some input until a given condition is met. You can then just call this function, changing the message and condition to be checked for. If you want to handle more than just string inputs, maybe you can make this function generic? `T <T> askUntilCondition(String message, Function<T, Boolean> condition)`. 

If you are confused about anything I've just said, ask in the comments and I'll explain further.